### PR TITLE
ENT-4863/3.12.x Set create permissions of monitord files in state directory to 0600

### PR DESCRIPTION
**Pull request by nickanderson**

### CODE REVIEW

**Comment by cf-bottom** [comment]
Thank you for submitting a pull request! Maybe @olehermanse can review this?

**Comment by nickanderson** [comment]
@cf-bottom jenkins, please

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/2473/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/2473/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-2473/

File: cf-monitord/env_monitor.c [hunk 1]
```diff
@@ -556,7 +557,10 @@ static void PublishEnvironment(Item *classes)
 
     unlink(ENVFILE_NEW);
 
-    if ((fp = fopen(ENVFILE_NEW, "a")) == NULL)
+    const mode_t old_umask = SetUmask(0077);
+    FILE *fp = safe_fopen(ENVFILE_NEW, "a");
```
Comment (vpodzime) [inline cf-monitord/env_monitor.c]: `FILE *fp;` five lines above has to be dropped.

Comment (nickanderson) [reply cf-monitord/env_monitor.c]: ACK

**Comment by nickanderson** [comment]
@cf-bottom jenkins, please

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/2474/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/2474/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-2474/

**Comment by nickanderson** [comment]
squashed

**Pull request merged**